docs/drafts/orchestration/boilerplate.md

```diff
@@ -0,0 +1,211 @@
+# Storefront Boilerplate
+
```

Comment:
In general we need to provide more explanation to the topic. 

I'd start with an introduction to explain what boilerplate is. 
> Boilerplate refers to the code that developers need to write in order to set up and configure the application before they can start building custom code. This includes the creating of the project directory structure, setup the dependencies and build the application entry point(s), such as index.html (CSR app) and index.js (SSR app).

Then we should outline the approach/vision of Oryx to boilerplate:
- Oryx tries to avoid boilerplate, as it is template code that cannot be easily updated to newer versions of Oryx
- All application logic is provided in packages which are distributed on npm. We should provide a link to the versioning strategy
- The feature sets, theming and configuration of the app is provided to the app builder
- Opinionated features are provided in presets. Presets can be added with a single line of code, and are provided to quickly setup a standard (demo) app. As soon as you start building a custom app, you might want to opt-out of the standard presets. Presets contain feature sets, themes and ...

We should provide the _ways_ to create the boilerplate code:
- manually, following steps below (I'm not too sure if we should do this, but we can. it means more maintenance on the docs)
- clone a repo (let's discuss in catch up on what repo we'll offer)
- (going forward we will provide a CLI to generate)

Then we describe what the boilerplate of Oryx entails and why:
- the bare minimum code: index.html, app.ts, ssr entry point (optional), ...

We could have a section for each of the main files and detail how they work, what they include.